sunbeam-python/sunbeam/core/watcher.py

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import time
+from typing import Any
+
+import timeout_decorator
+from watcherclient import v1 as watcher
+from watcherclient.common.apiclient.exceptions import NotFound
+from watcherclient.v1 import client as watcher_client
+
+from sunbeam.core.common import SunbeamException
+from sunbeam.core.juju import JujuHelper
+from sunbeam.core.openstack_api import get_admin_connection
+
+LOG = logging.getLogger(__name__)
+
+TIMEOUT = 60 * 3
+SLEEP_INTERVAL = 5
+ENABLE_MAINTENANCE_AUDIT_TEMPLATE_NAME = "Sunbeam Cluster Maintaining Template"
+ENABLE_MAINTENANCE_STRATEGY_NAME = "host_maintenance"
+ENABLE_MAINTENANCE_GOAL_NAME = "cluster_maintaining"
+
+WORKLOAD_BALANCING_GOAL_NAME = "workload_balancing"
+WORKLOAD_BALANCING_STRATEGY_NAME = "workload_stabilization"
+WORKLOAD_BALANCING_AUDIT_TEMPLATE_NAME = "Sunbeam Cluster Workload Balancing Template"
+
+
+def get_watcher_client(jhelper: JujuHelper) -> watcher_client.Client:
+    conn = get_admin_connection(jhelper=jhelper)
+    watcher_endpoint = conn.session.get_endpoint(
+        service_type="infra-optim",
+        # TODO: get region
```

Comment:
Are you going to fix it in the later PRs?

Reply:
Fixed